libcontainer/cgroups/fs2/fs2.go

```diff
@@ -301,3 +301,37 @@ func CheckMemoryUsage(dirPath string, r *configs.Resources) error {
 
 	return nil
 }
+
+func (m *Manager) MakeThreaded() error {
+	const (
+		cgTypeFile   = "cgroup.type"
+		typeThreaded = "threaded"
+	)
+
+	cgroupType, err := cgroups.ReadFile(m.dirPath, cgTypeFile)
+	if err != nil {
+		return err
+	}
+	cgroupType = strings.TrimSpace(cgroupType)
+
+	if cgroupType == typeThreaded {
+		return nil
+	}
+
+	err = cgroups.WriteFile(m.dirPath, cgTypeFile, typeThreaded)
+	if err != nil {
+		return err
+	}
+
+	cgroupType, err = cgroups.ReadFile(m.dirPath, cgTypeFile)
+	if err != nil {
+		return err
+	}
+	cgroupType = strings.TrimSpace(cgroupType)
+
+	if cgroupType != typeThreaded {
+		return fmt.Errorf("could not change cgroup type (%s) to %s", cgroupType, typeThreaded)
+	}
```

Comment:
I don't think reading back is required. The kernel returned no error, meaning the setting is applied.

Reply:
I've experienced many "silent errors" when dealing with cgroup files. If you're certain that the kernel will always produce an error then I'll be happy to remove it.

Reply:
And I'll be very interested for your thoughts and opinion regarding my [last comment](https://github.com/opencontainers/runc/pull/3691#issuecomment-1556190477)